### PR TITLE
feat(vscode): allow attaching development server to Dart debugger

### PR DESCRIPTION
**Pull request by alestiago**
## Status

**READY**

## Description

Allows attaching a development server to Dart's debugger via the command palette.

Changes:

- Registers debug-dev-server command

## Demonstration

https://github.com/VeryGoodOpenSource/dart_frog/assets/44524995/75f07036-d742-41f0-aa90-8c8e12b4a365

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: extensions/vscode/src/commands/debug-dev-server.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          "vscode.open",
+          Uri.parse(dartCodeMarketplaceUri)
+        );
+        return;
+      case "Cancel":
+      default:
+        return;
+    }
+    /* c8 ignore start */
+  }
```
Comment (alestiago) [inline extensions/vscode/src/commands/debug-dev-server.ts:37]: For some reason this closing } line is not covered by the coverage tooling. I have my suspicion that is related to ts transpilation to js and reporting back. I personally, don't think this is important for this Pull Request to land and I'm planning to create an issue to track this problem.

File: extensions/vscode/src/commands/debug-dev-server.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  /* c8 ignore stop */
+
+  if (!dartExtension.isActive) {
+    await window.withProgress(
+      {
+        location: 15,
+        title: `Activating Dart extension...`,
```
Comment (erickzanardo) [inline extensions/vscode/src/commands/debug-dev-server.ts:44]:
```suggestion
        title: 'Activating Dart extension...',
```

File: extensions/vscode/src/commands/debug-dev-server.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      },
+      async () => {
+        await dartExtension.activate();
+      }
```
Comment (erickzanardo) [inline extensions/vscode/src/commands/debug-dev-server.ts:48]:
```suggestion
      dartExtension.activate
```

File: extensions/vscode/src/commands/debug-dev-server.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+ *
+ * @param application The application to attach the debug session to.
+ */
+async function attachToDebugSession(
+  application: DartFrogApplication
+): Promise<void> {
+  await window.withProgress(
+    {
+      location: 15,
+      title: `Attaching to debug session...`,
```
Comment (erickzanardo) [inline extensions/vscode/src/commands/debug-dev-server.ts:116]:
```suggestion
      title: 'Attaching to debug session...',
```

**Review by erickzanardo — APPROVED** [review]

File: extensions/vscode/src/commands/debug-dev-server.ts (hunk 1 above)
Comment (alestiago) [reply extensions/vscode/src/commands/debug-dev-server.ts:37]: Tracking issue https://github.com/VeryGoodOpenSource/dart_frog/issues/961

**Pull request merged**